File: math/reverse-integer/reverseInteger.js
function reverseInteger(x) {
  /**
   * Reverses the digits of a signed 32-bit integer.
   * Returns 0 if the result overflows 32-bit signed integer range.
   */
  const minSigned32bit = -(2 ** 31);
  const maxSigned32bit = 2 ** 31 - 1;

  const sign = x < 0 ? -1 : 1;
  const reversedStr = Math.abs(x).toString().split("").reverse().join("");
  const reversedNum = sign * parseInt(reversedStr);

  if (!(minSigned32bit <= reversedNum && reversedNum <= maxSigned32bit)) {
    return 0;
  }

  return reversedNum;
}

// Example usage:
console.log(reverseInteger(-495)); // Output: -594
console.log(reverseInteger(1534236469)); // Output: 0 (overflow)
console.log(reverseInteger(123)); // Output: 321

File: math/reverse-integer/reverse_integer.py
def reverse_integer(x: int):
    min_signed_64bit = -2**31 - 1
    max_signed_64bit = 2**31 - 1

    stringed_number = str(x)
    negative = True if stringed_number[0] == "-" else False

    reverse_string = stringed_number[::-1] if not negative else stringed_number[:0:-1]

    converted_number = int(reverse_string) if not negative else int(reverse_string) * -1



    if not min_signed_64bit < converted_number < max_signed_64bit:
        #"Error: number not in the range of a Signed 32-bit."
        return -1

    return converted_number


print(reverse_integer(-495))
